CMakeLists.txt

```diff
@@ -101,19 +105,27 @@ elseif(ANDROID)
     target_sources(cpprealm PRIVATE src/cpprealm/internal/network/network_transport.cpp)
 elseif(MSVC)
     target_sources(cpprealm PRIVATE src/cpprealm/internal/network/network_transport.cpp)
-else()
+elseif(UNIX)
     find_package(CURL REQUIRED)
-    target_link_libraries(cpprealm PUBLIC
-            CURL::libcurl stdc++fs
-            )
-    target_sources(cpprealm PRIVATE src/cpprealm/internal/curl/network_transport.cpp)
+    if(NOT CURL_FOUND)
+        message(WARNING "CURL not found. Realm C++ will use internal networking components instead.")
+        target_sources(cpprealm PRIVATE src/cpprealm/internal/network/network_transport.cpp)
+    else()
+        target_link_libraries(cpprealm PUBLIC CURL::libcurl)
+        target_sources(cpprealm PRIVATE src/cpprealm/internal/curl/network_transport.cpp)
+    endif()
+    if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+        target_link_libraries(cpprealm PRIVATE stdc++fs)
+    endif()
+else()
+
 endif()
 
 if(MSVC)
     set_property(TARGET cpprealm PROPERTY
-      MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+      MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:${CMAKE_BUILD_TYPE}>:${CMAKE_BUILD_TYPE}>")
```

Comment:
Why change this? The possible values are either `MultiThreaded` or `MultiThreadedDebug`, the other possible values of `CMAKE_BUILD_TYPE` don't apply.